pkg/plugin/v2/api.go

```diff
@@ -158,7 +157,7 @@ func (p *createAPIPlugin) Validate() error {
 	return nil
 }
 
-func (p *createAPIPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
+func (p *createAPIPlugin) GetScaffolder() (cmdutil.Scaffolder, error) {
```

Comment:
@Adirio, 

It might cause problems for consumers. See:  `"sigs.k8s.io/kubebuilder/pkg/plugin/scaffold"` that is not internal and you are replacing for `"sigs.k8s.io/kubebuilder/pkg/plugin/internal/cmdutil"`. Could you please revert it?

Reply:
/lgtm cancel
/approve cancel

Reply:
`Scaffolder` was meant to be private, but it coulnd't be made private because it was being used from the `internal/cmdutil` package and from inside `pkg/plugin` (I was the one creating this interface). Now that we moved all into the same folder, we can make it private as it was intended.

The whole usage of this interface is private so no third-party library should be importing it.